torchchat/generate.py

```diff
@@ -593,7 +604,7 @@ def generate(
                     self.is_torchtune_model
                     or self.model.config.model_type == ModelType.Flamingo
                 ):
-                    model.setup_caches(max_batch_size=1, dtype=self.dtype)
+                    model.setup_caches(batch_size=1, dtype=self.dtype, encoder_max_seq_len=6404, decoder_max_seq_len=max_seq_length-1)
```

Comment:
Magic number 6404?

Reply:
it is the size that can hold single image input while one-gpu affordable.

Reply:
I will leave a comment